pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -179,11 +186,18 @@ func (cache *claimInfoCache) add(res *ClaimInfo) {
 	cache.claimInfo[res.ClaimName+res.Namespace] = res
 }
 
-func (cache *claimInfoCache) get(claimName, namespace string) *ClaimInfo {
+// getClaimInfo returns a deep copy of the cached claimInfo object
+// for the specified claim name and namespace
+func (cache *claimInfoCache) getClaimInfo(claimName, namespace string) (ClaimInfo, error) {
 	cache.RLock()
 	defer cache.RUnlock()
 
-	return cache.claimInfo[claimName+namespace]
+	claimInfo, ok := cache.claimInfo[claimName+namespace]
+	if !ok {
+		return ClaimInfo{}, fmt.Errorf("unable to get resource for namespace: %s, claim: %s", namespace, claimName)
```

Comment:
This should not contain the namespace or claimname in the error string. It is the responsibility of the caller to include any passed in parameters in its wrapper error string.

```suggestion
		return ClaimInfo{}, fmt.Errorf("claimInfo not found")
```

Reply:
Actually, thinking about this a bit more, should this just return a `bool` instead of an error?

Reply:
yes, that would be much clearer, I agreee.